go/database/mpt/nodes_test.go

```diff
@@ -6822,6 +6822,63 @@ func TestVisitPathToStorage_CanReachTerminalNodes(t *testing.T) {
 	}
 }
 
+func TestVisitPathToStorage_EmbeddedNode_Flag_Tracked(t *testing.T) {
+	address := common.Address{0xAB, 0xCD, 0xEF}
+	key := common.Key{0x12, 0x34, 0x56, 0x78}
+	var value common.Value
+	value[20] = 0x02
+	value[21] = 0x04
+
+	ctrl := gomock.NewController(t)
+	ctxt := newNodeContextWithConfig(t, ctrl, S5LiveConfig)
+
+	desc := &Extension{
+		path: AddressToNibblePath(address, ctxt)[0:30],
+		next: &Account{address: address, pathLength: 34, info: AccountInfo{Nonce: common.Nonce{0x01}, Balance: common.Balance{0x02}, CodeHash: common.Hash{0x03}},
+			storage: &Extension{
+				path:         KeyToNibblePath(key, ctxt)[0:40],
+				nextEmbedded: true,
+				next:         &Value{key: key, length: 24, value: value},
+			}},
+	}
+
+	ref, _ := ctxt.Build(desc)
+
+	var valueFound bool
+	visitor := NewMockNodeVisitor(ctrl)
+	visitor.EXPECT().Visit(gomock.Any(), gomock.Any()).DoAndReturn(func(node Node, info NodeInfo) VisitResponse {
+		var wantEmbedded bool
+		switch n := node.(type) {
+		case *AccountNode:
+			found, err := VisitPathToStorage(ctxt, &n.storage, key, visitor)
```

Comment:
If you visit another key with the same 40-byte prefix here (hard to generate because you would have to partially break keccak), the final "ValueNode" would not be indicated as embedded since `found` is false?